database/migrations/2019_01_16_174349_add_actions_table.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddActionsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('actions', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name', 255)->index()->comment('The action name of the post.');
+            $table->integer('campaign_id')->index()->comment('The campaign id the post is associated to.');
+            $table->string('post_type', 255)->index()->comment('The type of post the post is (e.g. photo, text, voter-reg, etc).');
+            $table->boolean('reportback')->comment('Whether or not the post should count as a reportback.');
+            $table->boolean('civic_action')->comment('Whether or not the post should count as a civic action.');
+            $table->boolean('scholarship_entry')->comment('Whether or not the post should count as a scholarship entry.');
+            $table->string('legacy_action_name', 255)->index()->nullable()->comment('The action from the posts table pre post metadata.');
```

Comment:
@DoSomething/team-bleed I'm not sure if we actually need this column? I'm thinking when we do the backfill, we can move what was in the `action` column in the `posts` table to just be the `name` in this table? More in this convo [here]( https://dosomething.slack.com/archives/C2BLZGBPH/p1547590807031100?thread_ts=1547579556.028900&cid=C2BLZGBPH).

Reply:
@chloealee yeah I think it makes sense to fill that information into the `name` column!

Reply:
That sounds reasonable. We'll just want to be careful not to mess w/ those names until we stop relying on them!